File: third-semester/Data-structure-and-algorithms-II/dynamic-programming/CuttingStick.py
def cortarVarilla(TamañoVarilla,listaPrecios):
    tabla= [0]*(TamañoVarilla+1)

    for Varilla_Actual in range (1,TamañoVarilla+1):
        mejor_valor=0
        
        print(tabla)
        for i in range (1,Varilla_Actual+1):
            valor=listaPrecios[i]+tabla[Varilla_Actual-i]#va buscar el precio de la varilla de longitud 2
            print(valor)
            if valor>mejor_valor:
                mejor_valor=valor
        tabla[Varilla_Actual]=mejor_valor
        print(tabla)

    return tabla [TamañoVarilla]
lista=[0,1,5,8,9,10]
cortarVarilla(4,lista)

File: 3-semester/Data-structure-and-algorithms-II/dynamic-programming/CuttingStick.java
public class CuttingStick {
    public static int cortarVarilla(int tamañoVarilla, int[] listaPrecios) {
        int[] tabla = new int[tamañoVarilla + 1];

        for (int varillaActual = 1; varillaActual <= tamañoVarilla; varillaActual++) {
            int mejorValor = 0;

            for (int i = 1; i <= varillaActual; i++) {
                int recorte1=listaPrecios[i];
                int recorte2 = tabla[varillaActual - i];
                int valor=recorte1+recorte2;
                if (valor > mejorValor) {
                    mejorValor = valor;
                }
            }

            tabla[varillaActual] = mejorValor;
        }

        return tabla[tamañoVarilla];
    }

    public static void main(String[] args) {
        int[] lista = {0, 1, 5, 8, 9, 10};
        int resultado = cortarVarilla(5, lista);
        System.out.println(resultado);
    }
}